File: mysql/mysql.js
class MySQL {
	constructor(opt={}) {
    this.mysql = require('mysql');
    require('dotenv').config();
    this.mdbh = this.mysql.createConnection({
      host: process.env.DBHOST,
      port: 3306,
      user: process.env.DBUSER,
      password: process.env.DBPASSWORD,
      database: 'mysql'
    });
  }

	dbconnect() {
    this.mdbh.connect(function (err) {
      if (err) {
          console.error('error connecting: ' + err.stack);
          return;
      }
    });
  }

	dbclose() {
    this.mdbh.end();
  }

  showSytemUsers() {
    const sql = "SELECT host,user,Select_priv FROM mysql.user";
    this.mdbh.query(sql, function (err, rows, fields) {
      if (err) { 
        console.log('err: ' + err);
      }
      for(let row of rows) {
        console.log([row.host, row.user, row.Select_priv].join("\t"));
      }
    });
  }
};
module.exports = MySQL;

if(!module.parent) {
	let rdb = new MySQL();
	rdb.dbconnect();
  rdb.showSytemUsers();
  rdb.dbclose();
}

File: mysql/mysql.rb
require 'dotenv'
require 'mysql2'

class MySQL
  def initialize(opt=[])
    Dotenv.load(
      File.join(File.dirname(File.expand_path(__FILE__)), '.env')
    )
  end

  def dbconnect
    @mdbh = Mysql2::Client.new(:host => ENV["DBHOST"], :username => ENV["DBUSER"], :password => ENV["DBPASSWORD"])
  end

  def showSystemUsers
    results = @mdbh.query("SELECT host,user,Select_priv FROM mysql.user")
    results.each do |row|
      print([row["host"], row['user'], row['Select_priv']].join("\t") + "\n")
    end
  end

  def dbclose
    @mdbh.close()
  end
end

if $0 == __FILE__
  rdb = MySQL.new()
  rdb.dbconnect()
  rdb.showSystemUsers()
  rdb.dbclose()
end
